src/main/thrift/parquet.thrift

```diff
@@ -211,6 +204,93 @@ struct Statistics {
    4: optional i64 distinct_count;
 }
 
+/** Empty structs to use as logical type annotations */
+struct StringType {}
```

Comment:
add a comment that it is UTF8 by default ? (in case we want to add an encoding field at some point).

Reply:
How about SQL engines that let the user put anything in a string without checking its encoding? Should they annotate strings with UTF8 even though they can not guarantee that users will actually store UTF8 in them, should they omit the annotation and thereby lose the information that those binary fields are actually strings, or should there be an encoding value for unknown string encoding? (See also: [IMPALA-5982](https://issues.apache.org/jira/browse/IMPALA-5982))

Reply:
SQL engines that store arbitrary binary shouldn't mark it as UTF-8 because it isn't guaranteed to be. Non-UTF-8 characters would cause exceptions in Java. There are two follow-ups to this:

1. Start validating that data written by Impala is UTF-8 and add the `UTF8` annotation to string columns. Scanning to verify an encoding shouldn't be too expensive.
2. Make sure that we can add annotations later in the expected schema. This is something we should add tests for in parquet-avro and other object models. It makes sense to me that converting one of these broken Parquet schemas to Avro shouldn't assume that binary columns can be read as Strings or Utf8 objects, but if I to read them that way by providing a schema with those fields as UTF8, I should be able to. Make sense?